website/blog/2024-11-25-announcing-lingui-5.0/index.md

```diff
@@ -0,0 +1,288 @@
+
+- `@lingui/react/macro` - for React (JSX) macros.
+- `@lingui/core/macro` - for Core (JS) macros.
+
+This allows developers to use Lingui macros in any JavaScript project without having to download the entire React package.
+
+### Whitespace Handling Changes
+
+Whitespace handling has been improved to provide greater consistency and meet developer expectations. Previously, Lingui's regex-based whitespace normalization in JSX macros sometimes produced unexpected results. With Lingui 5.0, Lingui now uses **JSX-specific whitespace rules**, producing predictable, clean output for React (JSX) macros.
+
+For JS macros, whitespace normalization has been **completely removed** based on developer feedback, leaving whitespace intact in expressions - ideal for text that requires special formatting, such as markdown.
+
+This change is breaking, meaning that some auto-generated message IDs may change. This could cause translations to lose their connection to the original message ID. To address this, you can use the translation memory capabilities of your translation management system (TMS) to restore translations for the updated message IDs.
+
+### New `useLingui` Macro
+
+In Lingui 5, we introduced a new [`useLingui`](/ref/macro#uselingui) macro that makes it easier to work with non-JSX messages within React components.
+
+Take the following code snippet as an example:
+
+```jsx
+import { t, msg } from "@lingui/macro";
+import { useLingui } from "@lingui/react";
+
+function MyComponent() {
+  const { i18n, _ } = useLingui();
+
+  const a = t(i18n)`Text`;
+}
+```
+
+With the new `useLingui` macro, you can simplify this code to:
+
+```jsx
+import { useLingui } from "@lingui/react/macro";
+
+function MyComponent() {
+  const { t } = useLingui();
+
+  const a = t`Text`;
+}
+```
+
+Under the hood, the `useLingui` macro is transpiled to the runtime [`useLingui`](/ref/react#uselingui) hook.
+
+### Standalone `babel-plugin-lingui-macro`
+
+There are now two ways to use Lingui macros with Babel in your project. You can choose between the `babel-macro-plugin` and the standalone `@lingui/babel-plugin-lingui-macro`.
+
+The new plugin offers faster transpiling and additional configuration options.
+
+We recommend replacing `babel-macro-plugin` with `@lingui/babel-plugin-lingui-macro` if you have direct access to your Babel configuration and don't use any other macros in your project.
+
+```diff title=".babelrc"
+{
+  "plugins": [
+-    "macros"
++    "@lingui/babel-plugin-lingui-macro"
+  ]
+}
+```
+
+### Print Placeholder Values for Better Translation Context
+
+Context is critical for translators (and AI) to provide accurate translations. This release introduces a new feature that prints placeholder values in PO comments.
+
+If the message contains unnamed placeholders such as `{0}`, Lingui will print their values in PO comments, giving translators and AI more context about what the placeholder is about.
```

Comment:
i'm wondering does "value" is a right word here? Sounds like lingui is going to print a real values of the variables, such as `{user} -> Timofei Iatsenko`